datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -187,6 +193,24 @@ func (f *ChainDataFetcher) Stop() error {
 func (f *ChainDataFetcher) sendRequests(startBlock, endBlock uint64, reqType cfTypes.RequestType, shouldUpdateCheckpoint bool, stopCh chan struct{}) {
 	logger.Info("sending requests is started", "startBlock", startBlock, "endBlock", endBlock)
 	for i := startBlock; i <= endBlock; i++ {
+		for { // spin lock if processing data size is larger than max
+			select {
+			case <-stopCh:
+				logger.Info("stopped making requests", "startBlock", startBlock, "endBlock", endBlock, "stoppedBlock", i)
+				return
+			default:
+			}
+
+			f.dataSizeLocker.RLock()
+			if f.processingDataSize <= f.maxProcessingDataSize {
+				f.dataSizeLocker.RUnlock()
+				break
+			}
+			f.dataSizeLocker.RUnlock()
+
+			logger.Debug("throttling the requests, sleeping", "interval", DefaultThrottlingInterval, "processingDataSize", f.processingDataSize, "maxDatasize", f.maxProcessingDataSize)
```

Comment:
I think we better increase this to WARN

Reply:
I have updated as you suggested (flag as well).